src/shared/sol-util-file.c

```diff
@@ -86,6 +86,10 @@ sol_util_write_file(const char *path, const char *fmt, ...)
     va_list ap;
     int ret;
 
+    ret = access(path, 0);
+    if (ret < 0)
+        return ret;
+
```

Comment:
For method 1st,  calling set_buffer_enable do not goto error when disabling buffer failed, just give warning to user and keep going instead of modifying sol_util_write_file function. What's your point?